crates/dkg-cli/README.md

```diff
@@ -0,0 +1,146 @@
+# DKG CLI Tool
+
+Command-line tool for Distributed Key Generation (DKG) and key rotation protocols.
+
+### Fresh DKG Steps
+
+1. Initization
+
+a. Initialize a committee, assuming an admin and members. 
+
+- The admin deploys its own `seal_ommittee` package.
+
+- Members to participate share their wallet addresses with the admin. This is the wallet to use to 
+complete the rest of the DKG onchain steps for registering and proposing. 
+
+- The admin initializes the Committee with a threshold and members addresses. This outputs the 
+committee object ID. Share `COMMITTEE_ID` with all members. 
+
+```bash
+SEAL_PKG=0x6af07f531232b02c14058d6f980169180d60eecfda69d5267cfabb282f42ff94
+COMMITTEE_PKG=0x434d924fd0b6d73b5de5791556a43820c98cb977396e4604e0166d9f3bb0bc8f
+
+ADDRESS_0=0x0636157e9d013585ff473b3b378499ac2f1d207ed07d70e2cd815711725bca9d
+ADDRESS_1=0xe6a37ff5cd968b6a666fb033d85eabc674449f44f9fc2b600e55e27354211ed6
+ADDRESS_2=0x223762117ab21a439f0f3f3b0577e838b8b26a37d9a1723a4be311243f4461b9
+
+sui client call --package $COMMITTEE_PKG --module seal_committee \
+  --function init_committee \
+  --args 2 "[\"$ADDRESS_0\", \"$ADDRESS_1\", \"$ADDRESS_2\"]"
+
+# share this with members. 
+COMMITTEE_ID=0xaa2e3cc1637ec725f3e4633f253d24e6f000f2c4a4c110f6c0bc38df1355b034
+```
+
+b. Members generate their ECIES and signing keypairs using CLI and export the environment variables.
+
+```bash
+cargo run --bin dkg-cli generate-keys
+
+export DKG_ENC_SK=$(grep enc_sk .dkg-keys | cut -d: -f2 | xargs)
+export DKG_SIGNING_SK=$(grep signing_sk .dkg-keys | cut -d: -f2 | xargs)
+
+export DKG_ENC_PK=0x...
+export DKG_SIGNING_PK=0x...
+```
+
+c. Members register the ECIES public key, signing public key and URL for the key server onchain.
+
+```bash
+sui client switch --address $ADDRESS_0 # your address
+sui client call --package $COMMITTEE_PKG --module seal_committee \
+  --function register \
+  --args $COMMITTEE_ID x"$DKG_ENC_PK" x"$DKG_SIGNING_PK" "<your_url>"
+```
+
+d. All members finished registration. Each member initialize the DKG protocol locally. 
+
+```bash
+cargo run --bin dkg-cli init \
+    --my-address $ADDRESS_0 \
+    --committee-id $COMMITTEE_ID \
+    --state-dir ./state
+```
+
+2. TODO: create message, process message and finalize DKG. 
```

Comment:
(reviewed until here)

Reply:
you can also review the init for key rotation steps below. 

the same command handles both. the difference is for rotation, the old_committee is fetched too for validation, TODO to build expected_pks. if continuing member, the --old-share arg is expected.

Reply:
I think we can make it a bit clearer if we split it a bit, e.g.,:

Coordinator:
- Deploys the contract if it's a new setup
- Initializes the committee and notifies members to run phase 1
- Watches the onchain state until all members registered
- Notifies all members to run phase 2
- Watches the offchain storage until all members posted their messages
- Notifies all members to run phase 3
- Monitor the chain to make sure all members finished (or handle issues if needed)

Member:
- Receive from coordinator the pkg and committee ids, check them
- Phase 1 - Generate keys and register them onchain
- Phase 2 - Create message and upload to offchain storage
- Phase 3 - Process all messages and update the committee

Reply:
they can be done in one command - let me combine them